src/main/java/online/partyrun/partyrunmatchingservice/domain/matching/service/MatchingService.java

```diff
@@ -53,4 +54,47 @@ private Mono<Matching> saveMatchAndSendEvents(List<MatchingMember> members, int
                                                     member.getId(), new MatchEvent(match));
                                         }));
     }
+
+    public Mono<Matching> setMemberStatus(
+            final Mono<String> member, final MatchingRequest request) {
+        return member.flatMap(
+                        mid ->
+                                matchingRepository
+                                        .findByMembersIdAndMembersStatus(
+                                                mid, MatchingMemberStatus.NO_RESPONSE)
+                                        .flatMap(
+                                                match ->
+                                                        matchingRepository
+                                                                .updateMatchingMemberStatus(
+                                                                        match.getId(),
+                                                                        mid,
+                                                                        MatchingMemberStatus
+                                                                                .getByIsJoin(
+                                                                                        request
+                                                                                                .isJoin()))
+                                                                .then(
+                                                                        Mono.defer(
+                                                                                () ->
+                                                                                        matchingRepository
+                                                                                                .findById(
+                                                                                                        match
+                                                                                                                .getId())))))
+                .flatMap(
+                        match -> {
+                            final boolean isUpdated = match.updateStatus();
+                            if (isUpdated) {
+                                return matchingRepository.save(match);
+                            }
+                            return Mono.just(match);
+                        })
+                .doOnSuccess(this::sendEvent);
```

Comment:
```suggestion
            .flatMap(match -> {
                            match.updateStatus();
                            if (!match.isWaiting()) {
                                sendEvent(match);
                                return matchingRepository.save(match);
                            }
                            return Mono.just(match);
                        })
```

updateStatus에서 변경과 조회를 같이하는게 좀 이상해서 바꿔봤습니다. 어떤가요?

Reply:
위에서도 언급했는데 현재 매칭 상황과 별개로 매치 상태 및 member 상태를 member들이 요청을 보낼 때 마다 이벤트로 반영하기로 클라이언트측과 합의했습니다

Reply:
또한 map 은 무언가를 변환하는 로직인데, 그 중간에 event를 전송한다는 변환로직과 관계없는 로직을 같이 포함하는거는 별로인 것 같네요